pulser-core/pulser/_hamiltonian_data/hamiltonian_data.py

```diff
@@ -583,12 +593,18 @@ def _interaction_matrix(self) -> "pm.torch.Tensor" | np.ndarray:
         return interactions
 
     @property
-    def noisy_interaction_matrix(self) -> pm.AbstractArray:
+    def noisy_interaction_matrices(self) -> list[pm.AbstractArray]:
+        """Get the noisy interaction matrix for each noise trajectory."""
+        return [x[0].interaction_matrix for x in self.noise_trajectories]
+
+    def _noisy_interaction_matrix(
+        self, register: BaseRegister, bad_atoms: dict
+    ) -> pm.AbstractArray:
         """Return the noisy interaction matrix."""
```

Comment:
I know this is a private method but I feel like the behavior is complex enough to deserve a docstring, especially on the return types